lib/vdsm/storage/lvm.py

```diff
@@ -604,6 +604,12 @@ def _reloadvgs(self, vgName=None):
                         "Marked vgs=%r as Unreadable due to reload failure",
                         logutils.Head(unreadable_vgs, max_items=20))
 
+                if len(vgNames) == 1 and not out:
+                    # Reload a specific VG name and failing without
+                    # an output might be indicative of a real error.
+                    raise se.VolumeGroupDoesNotExist.from_vg_name(
+                        vgName, str(error))
```

Comment:
Better to pass the error as is - this allow the caller to do later:

    except se.VolumeGroupDoesNotExist as e:
        if e.rc == ...

And it will work in the same way for formatting the error, since formatting will convert the error to a string.

Reply:
Uhm, then error should be `Optional[LVMCommandError]`?

In that case, if we want to support custom error messages as strings (was my idea anyway), we would need a different `reason` parameter. Otherwise, in reusing `error` we cannot assume that `e.error.rc` is available (it could be either `LVMCommandError` or `str` or `None`).

In any case, even with `error` and `reason`, we would still need to do something like:
```python
except se.VolumeGroupDoesNotExist as e:
    if e.error and e.error.rc == ...
```

Reply:
I was also thinking of adding this to the constructor:
```python
    assert error is None or isinstance(error, LVMCommandError)
```

Reply:
It makes sense and makes the code clear about the intent. In the worst case we will find that we need to pass another error in some case and we will make the type less strict.

There is one issue, assert statement may be removed in runtime if someone run this code with python -O option. So it is better to do:

    if error is not None and not isinstance(error, LVMCommandError):
        raise TypeError(
            f"Expecting instance of LVMCommandError, got {type(error)}")